src/main/java/com/github/princesslana/eriscasper/rx/websocket/RxWebSocket.java

```diff
@@ -29,31 +31,63 @@ public RxWebSocket(OkHttpClient http) {
             em -> {
               Request rq = new Request.Builder().url(url).build();
 
-              ws = http.newWebSocket(rq, new Listener(em));
+              ws = http.newWebSocket(rq, new Listener(RxWebSocket.this, em));
             })
+        .takeUntil(event -> closed)
         .doOnNext(e -> LOG.trace("Received: {}.", e))
         .doOnError(e -> LOG.warn("Error: {}.", e));
   }
 
+  private void close0() {
```

Comment:
Why `close0`? Does this method name make it obvious what this method does?

Reply:
it's the default closing of the socket private to the class. it's the ultimate goto when wanting to close the connection and stop accepting packets.